app/data/generators/personal-details.js

```diff
@@ -1,56 +1,151 @@
     dual: ['French', 'Swiss'],
     multiple: ['British', 'French', 'Swiss']
   }
+
   const selectedNationality = weighted.select({
     british: 0.65,
     irish: 0.05,
     french: 0.1,
     dual: 0.1,
     multiple: 0.1
   })
+
   const nationality = nationalities[selectedNationality]
 
+  // ---------------------------------------------------------------------------
+  // International candidate
   // Flag international candidate (does not have British/Irish nationality)
-  const isInternationalCandidate = !(nationality.includes('British') || nationality.includes('Irish'))
+  // ---------------------------------------------------------------------------
+  const isInternationalCandidate = !(nationality.includes('British')
+    || nationality.includes('Irish'))
 
-  const rightToWorkStudyOptions = {
-    yes: 'Yes',
-    unsure: 'Not yet, or not sure'
+  // ---------------------------------------------------------------------------
+  // Length of stay
+  // Lived in UK longer than 3 years
+  // ---------------------------------------------------------------------------
+  let lengthOfStay
+
+  if (!isInternationalCandidate) {
+
+    const lengthOfStayOptions = {
+      yes: 'Yes',
+      no: 'No'
+    }
+
+    const selectedLengthOfStay = weighted.select({
+      yes: 0.95,
+      no: 0.05
+    })
+
+    lengthOfStay = lengthOfStayOptions[selectedLengthOfStay]
   }
-  const selectedRightToWorkStudy = weighted.select({
-    yes: 0.6,
-    unsure: 0.4
-  })
-  const rightToWorkStudy = rightToWorkStudyOptions[selectedRightToWorkStudy]
 
-  const residency = {}
+  // ---------------------------------------------------------------------------
+  // Right to work or study
+  // ---------------------------------------------------------------------------
+  let rightToWorkStudy
+  let rightToWorkStudyHow
+  let rightToWorkStudyHowDetails
+
+  if (isInternationalCandidate) {
+
+    const rightToWorkStudyOptions = {
+      yes: 'Yes',
+      no: 'Not yet'
+    }
+
+    const selectedRightToWorkStudy = weighted.select({
+      yes: 0.6,
+      no: 0.4
+    })
+
+    rightToWorkStudy = rightToWorkStudyOptions[selectedRightToWorkStudy]
+
+    if (rightToWorkStudy === 'Not yet') {
+      const rightToWorkStudyHowOptions = {
+        visa: 'A visa sponsored by a course provider',
+        other: 'Another route'
+      }
+
+      const selectedRightToWorkStudyHow = weighted.select({
+        visa: 0.7,
+        other: 0.3
+      })
+
+      rightToWorkStudyHow = rightToWorkStudyHowOptions[selectedRightToWorkStudyHow]
+
+      if (rightToWorkStudyHow === 'Another route') {
+
+        let rightToWorkStudyHowChoices = [
+          'I am applying for a permanent residence card',
+          'I am applying for a visa',
+          'I have refugee status',
+          'I’m applying for a student visa',
+          'I’m applying for a spousal visa'
+        ]
+
+        let rightToWorkStudyHowDetails = rightToWorkStudyHowChoices.sort(() => 0.5 - Math.random())
```

Comment:
What does this do - perhaps a comment is needed

Reply:
Changed to `let rightToWorkStudyHowDetails = faker.helpers.randomize(rightToWorkStudyHowChoices)`